index.js

```diff
@@ -0,0 +1,60 @@
+
+var canvas = document.querySelector("canvas"), 
+ctx = canvas.getContext("2d");
+
+function draw(n,data){
+    let width = n, 
+    height = n, 
+    scale = canvas.width / n; 
+    fetch(data)
+        .then(response => response.json())
+        .then(cdata => {
+            
+            let makeColor = function(d) {return d;};
```

Comment:
looks like this assignment is not needed here, you can assign here null or remove it at all and make assignment in 'if' blocks